test/types/fastify.test-d.ts

```diff
@@ -18,3 +18,51 @@ expectType<LightMyRequestChain>(fastify({ http2: true, https: {}}).inject())
 
 expectError(fastify<http2.Http2Server>({ http2: false })) // http2 option must be true
 expectError(fastify<http2.Http2SecureServer>({ http2: false })) // http2 option must be true
+
+// server options
+fastify({ http2: true })
```

Comment:
Can these be wrapped with expectType or expectAssignable?

Reply:
Do you mean `expectType<FastifyInstance<http.Server, http.IncomingMessage, http.ServerResponse>>(fastify({ http2: true }))`? In the spirit of unit-testing I left that out as it's tested [elsewhere](https://github.com/fastify/fastify/pull/2277/files/5867d26d5faa08fa0839f62f14ee13c6ebb58b34#diff-7bd3eec77e6d6f08f2506c0170ce032eR10)

Reply:
Ahh okay I think you should simplify these tests into one:

```
expectAssignable<FastifyInstance>(fastify({
  // ... add as many options as possible here
})
```

Reply:
I split them up to guard against options that are accidentally defined as required:

```ts
interface ServerOptions {
  ignoreTrailingSlash?: boolean,
  connectionTimeout: number, // bad, should be `connectionTimeout?: number`
} 
```

In this case, a single test wouldn't catch that:

```ts
fastisfy({
  ignoreTrailingSlash: true,
  connectionTimeout: 1000
})
```

... and also unit-testing :)

Reply:
Okay then in this case I'd like to see 

```ts
expectAssignable<FastifyInstance>(fastify({ option: value })
```

for every option as this asserts the given instance is returning what we expect

Reply:
Cool. That works for every case except when any `logger` option is set (e.g. `expectAssignable<FastifyInstance>(fastify({ logger: true }))`), where it fails with: 

```text
Argument of type 'FastifyInstance<Server, IncomingMessage, ServerResponse, true>' is not assignable to parameter of type 'FastifyInstance<Server, IncomingMessage, ServerResponse, FastifyLoggerOptions<Server, IncomingMessage, ServerResponse>>'.
  Type 'true' is not assignable to type 'FastifyLoggerOptions<Server, IncomingMessage, ServerResponse>'
```

How can I fix that?

Reply:
Because the logger type is inferred by the generics, you'll need to use `expectType<FastifyInstance<http.server, http.incomingmessage, http.serverresponse, true>` to test this option

Reply:
Thanks for the help, fixed.